crates/gateway-config/src/extensions.rs

```diff
@@ -0,0 +1,67 @@
+use std::path::PathBuf;
+
+#[derive(PartialEq, serde::Deserialize, Debug, Clone)]
+#[serde(untagged)]
+pub enum ExtensionsConfig {
+    Version(String),
+    Structured(StructuredExtensionsConfig),
+}
+
+#[derive(PartialEq, serde::Deserialize, Debug, Clone)]
+pub struct StructuredExtensionsConfig {
+    pub version: String,
+    pub networking: bool,
+    pub stdout: bool,
+    pub stderr: bool,
+    pub environment_variables: bool,
+    pub max_pool_size: usize,
+}
+
+#[derive(Debug, Clone)]
+pub struct WasiExtensionsConfig {
```

Comment:
You could have added `HashMap<String, WasiExtensionConfig>` to the top level config and used a custom`impl Deserialize` with the enum to hide this logic.
I don't think I would have added the default location logic with `grafbase_extensions/...` here but in the wasi component loader IMHO. `{name}-{version}` sounds like a possible way to have name clashes, considering semver allows metadata information past the patch version. Would instead use `{name}/{version}` sounds a lot safer to me

Reply:
We will change this completely as discussed in our call today. Not in this PR though.

Reply:
It will be done with this issue:

https://linear.app/grafbase/issue/GB-8373/extension-loading-in-the-gateway-reload